packages/grid/data-grid/src/tests/toolbar.DataGrid.test.tsx

```diff
@@ -73,4 +73,32 @@ describe('<DataGrid /> - Toolbar', () => {
       maxHeight: `${Math.floor(rowHeight * COMFORTABLE_DENSITY_FACTOR)}px`,
     });
   });
+
+  it('should increase grid density even if toolbar is not enabled', () => {
+    const rowHeight = 30;
+    render(
+      <div style={{ width: 300, height: 300 }}>
+        <DataGrid {...baselineProps} rowHeight={rowHeight} density="compact" />
+      </div>,
+    );
+
+    // @ts-expect-error need to migrate helpers to TypeScript
+    expect(document.querySelector('.MuiDataGrid-row')).toHaveInlineStyle({
```

Comment:
Accessibility tree?

```suggestion
    expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
```

Reply:
It does look better that way. I selected it by class because I wanted to skip the header and avoid selecting an element in the array but I like using the `role` more.